custom/icds_reports/utils/aggregation_helpers/distributed/agg_dashboard_activity.py

```diff
@@ -141,12 +141,17 @@ def create_table_query(self):
 
     def get_last_agg_date(self):
         from custom.icds_reports.models.aggregate import DashboardUserActivityReport
+
+        #because we dont expect the report to fail for 7 consecutive days
```

Comment:
dangerous assumption, is there a way to trigger something if this does happen? at least some kind of alert. we had similar issues with the warehouse in the past, where it failed for so long that all the data was broken but no one knew because the task kept running.

Reply:
There is no trigger set up yet for this. The caveat here is that even when it fails for 7 days consecutively. The 8th day it will just the query with the default old date which will still give the correct data but yes surely query will take longer to finish.

I can make a followup PR to set some alert or failure email.